### ISSUE TITLE
Time issues cause trakt sync to crash 

### ISSUE THREAD

**Issue by SurfingSpider** [Razzeee/script.trakt#166]
I am also having problems related to time I think. I have also seen a couple of other posts about it on the kodi forums .I was hoping you could point me in the right direction.

I have seen Razee ask someone with a related issue (on the kodi forum).... 
```Please check your system time and are you using mysql or edited the sqlite library? Do you have anything collected or watched with a year outside of 1970 to 2038? What's your timezone?).```

I am running win7, kodi 14.2 rc & MySQL setup all clocks are synced to GMT. I have Trakt addon v3.0.0 installed. I also have tv series from 1940 (tom & jerry, 5 year old loves it)

Here is the relevant section of my logs...
```
19:53:54 T:3984   ERROR: Exception in thread trakt-sync:
                                            Traceback (most recent call last):
                                              File "C:\Program Files\Kodi\system\python\Lib\threading.py", line 810, in __bootstrap_inner
                                                self.run()
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\service.py", line 281, in run
                                                sync.sync()
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\sync.py", line 945, in sync
                                                self.__syncEpisodes()
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\sync.py", line 514, in __syncEpisodes
                                                kodiShowsCollected, kodiShowsWatched = self.__kodiLoadShows()
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\sync.py", line 86, in __kodiLoadShows
                                                show['seasons'] = utilities.kodiRpcToTraktMediaObjects(data)
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\utilities.py", line 390, in kodiRpcToTraktMediaObjects
                                                episodeObject = kodiRpcToTraktMediaObject('episode', episode, mode)
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\utilities.py", line 336, in kodiRpcToTraktMediaObject
                                                episode['watched_at'] = convertDateTimeToUTC(data['lastplayed'])
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.trakt\utilities.py", line 420, in convertDateTimeToUTC
                                                utc = local.astimezone(tzutc())
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.module.dateutil\lib\dateutil\tz.py", line 113, in utcoffset
                                                if self._isdst(dt):
                                              File "C:\Users\Administrator\AppData\Roaming\Kodi\addons\script.module.dateutil\lib\dateutil\tz.py", line 157, in _isdst
                                                return time.localtime(timestamp+time.timezone).tm_isdst
                                            ValueError: (22, 'Invalid argument')
```
At the moment if you run a manual sync it runs through all the movies fine, but gets stuck at... "Loading episode data from kodi, Parsing 3 of 44 episode data from kodi". After that I have to crash out of kodi

any help will be much appreciated. Please forgive any ignorance on my part I don't by a long stretch call myself a programmer more an educated user.

**Comment by nate1280** [comment]
Hmm, thats a new one.

The traceback makes no sense, can't really determine where its occuring.  Some of the lines its going from have nothing to do with one another, which is why it makes no sense.

UTF-8 decode errors usually mean there's problems with show names or episode names, and are a pain to try and catch because they happen so infrequently.

Are you using mysql or sqlite to store your shows/movies?

**Comment by ezfranca** [comment]
Well guys, just to update. After a did the update ( from the openelec build to this one, from the master ) I used the convmv to convert all my files to utf8. I removed some files with strange characters and did a full cleanup. 
After that, it worked. It was executed until the end. I think there was some file that had some weird encoded character. I think a good way to handle it is a error handling that shows up which file the plugin is processing on the time of the error, so we could simple delete or convert it to the correct charset.

for the question, I am using sqlite, but I am planning to move to mysql soon. Any considerations to do it?

This addon is amazing! The only thing I would suggest is better error handling , to show us which file or which dir has something wrong.

**Comment by nate1280** [comment]
Glad to hear you got it working.

The database was just a curiosity, I use MySQL, so all my testing/debugging is done in that environment.

I've tried to add in as much error handling as possible, but some stuff still slips through the cracks.

Unfortunately, it doesn't do a file at a time, the json request return large chunks of data and the problem could be anywhere within that data stream, so there's no real easy way to pinpoint it.

And like I said in the previous post, the error that did occur, the traceback makes no sense to me, it seems to jump all over and between unrelated code and code that isn't even a function but just a straight variable assignment (the self._runSilent line for example).

**Comment by ezfranca** [comment]
There is a way to handle the error just putting the last or the most recent entry of json ? I know it is not your fault, but it is impossible to track which file is breaking the sync. 
Or, the sync could just move on after a error like that, because now, if I get this error during a manual sync, I need to restart the computer, because I cannot even cancel it.
:(
and congratz for the amazing job here :)

**Comment by nate1280** [comment]
The json is a full dump, and the error occurs in the decoding of the json.  The xbmc api returns it in string format, then it gets exploded into json format to work with, its in there that the error occurs (in the python simplejson).

Again, I can't determine where the error actually occurs from your trace, there is 1 check in the function used already which should catch most stuff but not sure where the above error is really occuring.

**Comment by ezfranca** [comment]
Well, now I got it. I did not know the return is a string. I thought the error occurs when the data was already parsed into a structured data, not string.

I think, in this case, the best way is a faq about the weird charsets encode that you have from not so reliable torrent sources :)

One more time, thanks a lot!

**Comment by Razzeee** [comment]
Well it's not about shows or movies that are from before 1970.
It's about movies/episodes where you have set the 'lastplayed' or 'dateadded' fields in your mysql to something outside of 1970 to 2038
So thats what we hould look for, it's pretty easy to query this from mysql (at least if you got one right before you)

**Comment by SurfingSpider** [comment]
Success!

I found that about 10 episodes had entries that were 1969-12-31 00:00:00 in the lastPlayed field for some reason. I have moved the machine that SQL database was installed on since watching those episodes and reimported from separate exported NFO's. There must have been a short period where the data was somehow getting corrupted.

I just went to those episodes within kodi and skipped to the end of each of them again to reset the lastPlayed field.

Many thanks for your time. looking into my issue
